src/matlab2tikz.m

```diff
@@ -518,19 +531,19 @@ function matlab2tikz(varargin)
     % Finally print it to the file
     addComments(fid, m2t.content.comment);
     addStandalone(m2t, fid, 'preamble');
-    addCustomCode(fid, '', m2t.cmdOpts.Results.extraCode, '');
+    addCustomCode(fid, '', m2t.args.extraCode, '');
     addStandalone(m2t, fid, 'begin');
 
     printAll(m2t, m2t.content, fid); % actual plotting happens here
 
-    addCustomCode(fid, '\n', m2t.cmdOpts.Results.extraCodeAtEnd, '');
+    addCustomCode(fid, '\n', m2t.args.extraCodeAtEnd, '');
 
     addStandalone(m2t, fid, 'end');
 end
 % ==============================================================================
 function addStandalone(m2t, fid, part)
     % writes a part of a standalone LaTeX file definition
-    if m2t.cmdOpts.Results.standalone
+    if m2t.args.standalone
```

Comment:
I really like it, but please open a feature request for that (maybe with the current changes you have). I suppose it would be enough to add a parameter `preamble` to our input arguments, change a few references here or there and done?